netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java

```diff
@@ -195,7 +241,19 @@ public ChannelFuture writeSettingsAck(ChannelHandlerContext ctx, ChannelPromise
   @Override
   public void close() {
     super.close();
-    cancelPendingStreams();
+
+    if (!closed) {
+      closed = true;
+
+      // Fail all buffered streams.
+      ChannelClosedException e = new ChannelClosedException();
+      PendingStream[] streams = pendingStreams.values().toArray(
```

Comment:
Am I correct in reading the code, that this copy is unnecessary and this loop operates basically the same as the removed `cancelPendingStreams()`?